### PR TITLE
Early return when module aliases are not defined

### PR DESCRIPTION
**Pull request by elsh**
Move stack push out of a nested loop
Resolves rdar://89776259

### CODE REVIEW

**Comment by elsh** [comment]
@swift-ci smoke test

**Comment by tomerd** [comment]
@swift-ci smoke test

File: Sources/PackageGraph/PackageGraph+Loading.swift [hunk 1]
```diff
@@ -265,7 +265,7 @@ private func createResolvedPackages(
     }
 
     // Gather all module aliases specified for targets in all dependent packages
-    let pkgToAliasesMap = gatherModuleAliases(for: rootManifests.first?.key, with: packagesByIdentity, onError: observabilityScope)
+    let pkgToAliasesMap = gatherModuleAliases(from: packageBuilders, for: rootManifests.first?.key, with: packagesByIdentity, onError: observabilityScope)
```
Comment (tomerd) [inline Sources/PackageGraph/PackageGraph+Loading.swift:268]: nit: use clearer variable names, pkgToAliasesMap -> packageAliases

Comment (tomerd) [reply Sources/PackageGraph/PackageGraph+Loading.swift:268]: onError -> observabilityScope

Comment (tomerd) [reply Sources/PackageGraph/PackageGraph+Loading.swift:268]: question: is `for: rootManifests.first?.key` a problem? what If there are multiple root packages, or is that not implemented yet?

File: Sources/PackageGraph/PackageGraph+Loading.swift [hunk 2]
```diff
@@ -580,14 +591,13 @@ private func walkPkgTreeAndGetModuleAliases(for pkgID: PackageIdentity,
...
-            // pkgID here, otherwise need pkgID != pkgInChain check
-            // in the for loop above
-            pkgStack.append(pkgID)
         }
-        
+        // Add pkgID to a stack used to keep track of multiple
+        // aliases specified in the package chain. Need to add
+        // pkgID here, otherwise need pkgID != pkgInChain check
+        // in the for loop above
+        pkgStack.append(pkgID)
```
Comment (tomerd) [inline Sources/PackageGraph/PackageGraph+Loading.swift:599]: this change seems unrelated but potentially material?

**Comment by tomerd** [comment]
@swift-ci please smoke test linux

File: Sources/PackageGraph/PackageGraph+Loading.swift (hunk 2 above)
Comment (elsh) [reply Sources/PackageGraph/PackageGraph+Loading.swift:599]: This change makes sure pkgID is only added once, so it's an improvement.

**Comment by elsh** [comment]
@swift-ci smoke test

**Comment by elsh** [comment]
@swift-ci smoke test macOS

**Comment by elsh** [comment]
@swift-ci smoke test macOS

**Comment by tomerd** [comment]
@swift-ci please smoke test macOS

**Review by tomerd — APPROVED** [review]

**Pull request merged**